contracts/contracts/strategies/NativeStaking/NativeStakingSSVStrategy.sol

```diff
@@ -0,0 +1,264 @@
+        uint256 ethRewards = executionRewards + consensusRewards;
+
+        require(
+            address(this).balance >= ethRewards,
+            "insufficient eth balance"
+        );
+
+        if (ethRewards > 0) {
+            // reset the counter keeping track of beacon chain consensus rewards
+            consensusRewards = 0;
+
+            // Convert ETH rewards to WETH
+            IWETH9(WETH_TOKEN_ADDRESS).deposit{ value: ethRewards }();
+
+            emit RewardTokenCollected(
+                harvesterAddress,
+                WETH_TOKEN_ADDRESS,
+                ethRewards
+            );
+            IERC20(WETH_TOKEN_ADDRESS).safeTransfer(
+                harvesterAddress,
+                ethRewards
+            );
+        }
+    }
+
+    /// @notice Unlike other strategies, this does not deposit assets into the underlying platform.
+    /// It just checks the asset is WETH and emits the Deposit event.
+    /// To deposit WETH into validators `registerSsvValidator` and `stakeEth` must be used.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    /// @param _asset Address of asset to deposit. Has to be WETH.
+    /// @param _amount Amount of assets that were transferred to the strategy by the vault.
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        require(_asset == WETH_TOKEN_ADDRESS, "Unsupported asset");
+        _deposit(_asset, _amount);
+    }
+
+    /// @dev Deposit WETH to this strategy so it can later be staked into a validator.
+    /// @param _asset Address of WETH
+    /// @param _amount Amount of WETH to deposit
+    function _deposit(address _asset, uint256 _amount) internal {
+        require(_amount > 0, "Must deposit something");
+        /*
+         * We could do a check here that would revert when "_amount % 32 ether != 0". With the idea of
+         * not allowing deposits that will result in WETH sitting on the strategy after all the possible batches
+         * of 32ETH have been staked.
+         * But someone could mess with our strategy by sending some WETH to it. And we might want to deposit just
+         * enough WETH to add it up to 32 so it can be staked. For that reason the check is left out.
+         *
+         * WETH sitting on the strategy won't interfere with the accounting since accounting only operates on ETH.
+         */
+        emit Deposit(_asset, address(0), _amount);
+    }
+
+    /// @notice Unlike other strategies, this does not deposit assets into the underlying platform.
+    /// It just emits the Deposit event.
+    /// To deposit WETH into validators `registerSsvValidator` and `stakeEth` must be used.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 wethBalance = IERC20(WETH_TOKEN_ADDRESS).balanceOf(
```

Comment:
I just realized that calling `depositAll` will communicate incorrect values since one can deposit WETH and that WETH will stay on the contract until the defender action stakes it. 

If `depositAll` is called again before all WETH is unwrapped and staked this value will be too high, since it will include the WETH that was on the contract before the Vault sent additional WETH to it. This might be misleading for the monitoring tools if they rely on `Deposit` events instead of calling `checkBalance`.

Reply:
That's true, but what's the alternative? Store the deposited WETH balance in storage so the actual deposit amount can be calculated?

Reply:
Yeah I can't think of a better alternative than that. And adding so much complexity doesn't seem to justify the correct values in the Deposit events.